2017 Robot/src/org/usfirst/frc/team4999/lights/Packet.java

```diff
@@ -0,0 +1,162 @@
+package org.usfirst.frc.team4999.lights;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A command sent to the arduino to light up a pixel or area of pixels with a certain color.
+ * Follows <a href="https://github.com/momentumfrc/2017Steamworks/wiki/Light-strings-on-the-Robot">this</a> structure.
+ * @author jordan
+ *
+ */
+public class Packet {
+	// The max size of a packet is 16 bytes
+	// Before a packet is sent, B_BUFFER is filled with that packet's data. B_BUFFER is then passed to the I2C writeBulk() method
+	private static final ByteBuffer B_BUFFER = ByteBuffer.allocateDirect(16);
+	
+	// All according to specification
+	private final byte DISPLAY_FRAME = 0x01;
+	private final byte SET_SINGLE = 0x02;
+	private final byte SET_AREA = 0x03;
+	private final byte SET_STRIDE = 0x04;
+	
+	// No point in constantly recreating the show packet since its the same every time
+	private static Packet show;
+	
+	private byte command;
+	private byte[] data;
+	
+	/**
+	 * Creats a DISPLAY_FRAME packet
+	 */
+	private Packet() {
```

Comment:
This could be handled a lot more cleanly.  My recommendation would be to use the Factory pattern to generate Packets.  The Packet class itself shouldn't have any knowledge of how to make specific types of Packets, nor should particular constructor signatures magically map to commands.

I can help flesh this out if you're not familiar with this pattern.